### PR TITLE
Add xunit.netcore.extensions

### PR DESCRIPTION
**Pull request by ViktorHofer**
Port from https://github.com/dotnet/buildtools/tree/master/src/xunit.netcore.extensions

Also moved versions from the GitSyncClient to versions.props as discussed with @Anipik and fixed projects outside of the src directory in the sln.

### CODE REVIEW

File: src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj [hunk 1]
```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <OutputType>Library</OutputType>
+    <AssemblyName>xunit.netcore.extensions</AssemblyName>
+    <IsPackable>true</IsPackable>
+    
+    <PackageId>Microsoft.xunit.netcore.extensions</PackageId>
```
Comment (tmat) [inline src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj]: >Microsoft.xunit.netcore.extensions [](start = 15, length = 34)

The capitalization looks odd. Why not `Microsoft.XUnit.NetCore.Extensions`?

File: src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj [hunk 2]
```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <OutputType>Library</OutputType>
+    <AssemblyName>xunit.netcore.extensions</AssemblyName>
```
Comment (tmat) [inline src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj]: >AssemblyName>xunit.netcore.extensions</AssemblyName> [](start = 5, length = 52)

Why not `Microsoft.XUnit.NetCore.Extensions`?

File: src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    <IsPackable>true</IsPackable>
+    
+    <PackageId>Microsoft.xunit.netcore.extensions</PackageId>
+    <Authors>Microsoft</Authors>
+    <VersionPrefix>2.4.0</VersionPrefix>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
```
Comment (tmat) [inline src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj]: ><PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance> [](start = 4, length = 71)

Not needed, this is set to true by default by the toolset.

File: src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj [hunk 4]
```diff
@@ -0,0 +1,24 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <OutputType>Library</OutputType>
+    <AssemblyName>xunit.netcore.extensions</AssemblyName>
+    <IsPackable>true</IsPackable>
+    
+    <PackageId>Microsoft.xunit.netcore.extensions</PackageId>
+    <Authors>Microsoft</Authors>
```
Comment (tmat) [inline src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj]: ><Authors>Microsoft</Authors> [](start = 4, length = 28)

Not needed, this is set by default by the toolset.

File: src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    <VersionPrefix>2.4.0</VersionPrefix>
+    <PackageRequireLicenseAcceptance>true</PackageRequireLicenseAcceptance>
+    <Description>This package provides things like various traits and discovers like OuterLoop/ActiveIssue that are used by .NET Core framework test projects.</Description>
+    <Copyright>Copyright © Microsoft Corporation</Copyright>
+    <PackageLicenseUrl>http://go.microsoft.com/fwlink/?LinkId=329770</PackageLicenseUrl>
+    <PackageProjectUrl>https://github.com/dotnet/arcade</PackageProjectUrl>
+    <PackageIconUrl>http://go.microsoft.com/fwlink/?LinkID=288859</PackageIconUrl>
```
Comment (tmat) [inline src/xunit.netcore.extensions/src/Xunit.NetCore.Extensions.csproj]: All these are preset.

File: src/xunit.netcore.extensions/src/DesktopTraceListener.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+
+namespace Xunit.NetCore.Extensions
```
Comment (tmat) [inline src/xunit.netcore.extensions/src/DesktopTraceListener.cs]: >Xunit.NetCore.Extensions [](start = 10, length = 24)

Can we change the namespace to Microsoft.DotNet.XUnitExtensions?

**Comment by ViktorHofer** [comment]
Thanks! I decided to use your suggestion `Microsoft.DotNet.XUnitExtensions` for everything.

**Comment by rainersigwald** [comment]
Hello, 👋, it is me, the guy who always whines about importing the full history of a project if possible.

Do you think you could import the full history of this project? It helps with understanding decisions made pre-port, and it will probably reduce pressure on you in the future to answer questions about this code.

**Comment by ViktorHofer** [comment]
Can you please clarify what you mean with "importing the full history of this project"?

**Comment by ViktorHofer** [comment]
Ah sorry, I thought this message was in another repo. Of course.

**Pull request closed without merging**